src/cmap/conn/mod.rs

```diff
@@ -38,10 +38,15 @@ pub(crate) use wire::next_request_id;
 
 /// User-facing information about a connection to the database.
 #[derive(Clone, Debug)]
+#[non_exhaustive]
```

Comment:
it seems we did not mark this type non-exhaustive before, and I'm not sure what our current thinking is on retroactively adding it and adding more properties to the type, but I figured I'd start with this and get your opinions. 
if we feel like we should avoid doing this, I could add a new field `server_connection_id` directly to the monitoring events rather than sticking it on `ConnectionInfo`, and we could update the API in 3.0.

Reply:
It looks like we don't actually expose the `ConnectionInfo` struct in our public API -- it's not exported in lib.rs, so it doesn't show up in documentation and users can't import the type. For that reason I'm not sure that this would actually be a breaking change? `#[non_exhuastive]` is needed to prevent constructing/destructuring, but I don't think that's possible to begin with if you're not able to import the type.

Reply:
oh interesting. well, it seems like we probably do want to expose it in our API so users can see the available fields in the docs. would we want to expose it via the `cmap` module (which currently has nothing else public) or just re-export it via `pub use` from the `event` module?

Reply:
I'd be in favor of `event`, we've done that for other similar types related to SDAM/CMAP monitoring.

Reply:
added a `pub use` to make it visible in `event::command`